File: simulate.py
import requests
import json
import random
import time

SUPABASE_URL = 'https://yxzzfqyehllogzzhdtmc.supabase.co'
SUPABASE_ANON_KEY = 'sb_publishable_wWVQ59Rejod5Oc1X4s_eeQ_ONbXzyi2'

headers = {
    'apikey': SUPABASE_ANON_KEY,
    'Authorization': f'Bearer {SUPABASE_ANON_KEY}',
    'Content-Type': 'application/json',
    'Prefer': 'return=representation'
}

def get_random_offset():
    return (random.random() - 0.5) * 0.04

def start_simulation():
    print("Iniciando simulación...")

    base_lat = -17.3895
    base_lng = -66.1568

    # 1. Crear 20 pedidos falsos
    pedidos = []
    for i in range(1, 21):
        pedidos.append({
            'cliente_nombre': f'Cliente Sim {i}',
            'whatsapp': f'700000{i:02d}',
            'producto': 'Garrafa GLP 10kg',
            'lat': base_lat + get_random_offset(),
            'lng': base_lng + get_random_offset(),
            'estado': 'buscando',
            'ciudad': 'cochabamba'
        })

    pedidos_result = []
    try:
        res = requests.post(f"{SUPABASE_URL}/rest/v1/pedidos_activos", headers=headers, json=pedidos)
        if res.status_code in (200, 201):
            pedidos_result = res.json()
            print(f"Creados {len(pedidos_result)} pedidos simulados.")
        else:
            print(f"Error creando pedidos: {res.text}")
    except Exception as e:
        print(f"Excepcion creando pedidos: {e}")

    # 2. Crear 5 repartidores
    choferes = []
    for i in range(1, 6):
        choferes.append({
            'nombre_completo': f'Repartidor Sim {i}',
            'telefono_whatsapp': f'6000000{i}',
            'placa': f'SIM-{i}00',
            'lat': base_lat + get_random_offset(),
            'lng': base_lng + get_random_offset(),
            'is_active': True,
            'ciudad': 'cochabamba'
        })

    choferes_result = []
    try:
        res = requests.post(f"{SUPABASE_URL}/rest/v1/choferes_habilitados", headers=headers, json=choferes)
        if res.status_code in (200, 201):
            choferes_result = res.json()
            print(f"Creados {len(choferes_result)} choferes simulados.")
        else:
            print(f"Error creando choferes: {res.text}")
    except Exception as e:
        print(f"Excepcion creando choferes: {e}")

    # 3. Simular movimiento por 5 minutos (12 iteraciones de 25 segundos)
    print("Simulando movimiento...")
    for step in range(12):
        time.sleep(25)
        print(f"Movimiento paso {step + 1}/12...")
        
        for chofer in choferes_result:
            chofer['lat'] += (random.random() - 0.5) * 0.002
            chofer['lng'] += (random.random() - 0.5) * 0.002
            
            try:
                requests.patch(f"{SUPABASE_URL}/rest/v1/choferes_habilitados?id=eq.{chofer['id']}", 
                               headers=headers, 
                               json={'lat': chofer['lat'], 'lng': chofer['lng']})
            except Exception as e:
                pass

    # 4. Borrar todo
    print("Limpiando simulación...")
    for pedido in pedidos_result:
        requests.delete(f"{SUPABASE_URL}/rest/v1/pedidos_activos?id=eq.{pedido['id']}", headers=headers)
    for chofer in choferes_result:
        requests.delete(f"{SUPABASE_URL}/rest/v1/choferes_habilitados?id=eq.{chofer['id']}", headers=headers)
    
    print("Simulación finalizada.")

if __name__ == '__main__':
    start_simulation()

File: simulate.js
const SUPABASE_URL = 'https://yxzzfqyehllogzzhdtmc.supabase.co';
const SUPABASE_ANON_KEY = 'sb_publishable_wWVQ59Rejod5Oc1X4s_eeQ_ONbXzyi2';

const headers = {
  'apikey': SUPABASE_ANON_KEY,
  'Authorization': `Bearer ${SUPABASE_ANON_KEY}`,
  'Content-Type': 'application/json',
  'Prefer': 'return=representation'
};

function getRandomOffset() {
  return (Math.random() - 0.5) * 0.04;
}

async function startSimulation() {
  console.log("Iniciando simulación...");

  const baseLat = -17.3895;
  const baseLng = -66.1568;

  // 1. Crear 20 pedidos falsos
  const pedidos = [];
  for (let i = 1; i <= 20; i++) {
    pedidos.push({
      cliente_nombre: `Cliente Sim ${i}`,
      whatsapp: `700000${i.toString().padStart(2, '0')}`,
      producto: 'Garrafa GLP 10kg',
      lat: baseLat + getRandomOffset(),
      lng: baseLng + getRandomOffset(),
      estado: 'buscando',
      ciudad: 'cochabamba'
    });
  }

  let pedidosResult;
  try {
    const res = await fetch(`${SUPABASE_URL}/rest/v1/pedidos_activos`, {
      method: 'POST',
      headers,
      body: JSON.stringify(pedidos)
    });
    pedidosResult = await res.json();
    console.log(`Creados ${pedidosResult.length} pedidos simulados.`);
  } catch(e) {
    console.error("Error creando pedidos", e);
  }

  // 2. Crear 5 repartidores
  const choferes = [];
  for (let i = 1; i <= 5; i++) {
    choferes.push({
      nombre_completo: `Repartidor Sim ${i}`,
      telefono_whatsapp: `6000000${i}`,
      placa: `SIM-${i}00`,
      lat: baseLat + getRandomOffset(),
      lng: baseLng + getRandomOffset(),
      is_active: true,
      ciudad: 'cochabamba'
    });
  }

  let choferesResult;
  try {
    const res = await fetch(`${SUPABASE_URL}/rest/v1/choferes_habilitados`, {
      method: 'POST',
      headers,
      body: JSON.stringify(choferes)
    });
    choferesResult = await res.json();
    console.log(`Creados ${choferesResult.length} choferes simulados.`);
  } catch(e) {
    console.error("Error creando choferes", e);
  }

  // 3. Simular movimiento por 5 minutos (12 iteraciones de 25 segundos)
  console.log("Simulando movimiento...");
  for (let step = 0; step < 12; step++) {
    await new Promise(r => setTimeout(r, 25000));
    console.log(`Movimiento paso ${step + 1}/12...`);
    
    for (const chofer of choferesResult) {
      chofer.lat += (Math.random() - 0.5) * 0.002;
      chofer.lng += (Math.random() - 0.5) * 0.002;
      
      await fetch(`${SUPABASE_URL}/rest/v1/choferes_habilitados?id=eq.${chofer.id}`, {
        method: 'PATCH',
        headers,
        body: JSON.stringify({ lat: chofer.lat, lng: chofer.lng })
      });
    }
  }

  // 4. Borrar todo
  console.log("Limpiando simulación...");
  for (const pedido of pedidosResult) {
    await fetch(`${SUPABASE_URL}/rest/v1/pedidos_activos?id=eq.${pedido.id}`, { method: 'DELETE', headers });
  }
  for (const chofer of choferesResult) {
    await fetch(`${SUPABASE_URL}/rest/v1/choferes_habilitados?id=eq.${chofer.id}`, { method: 'DELETE', headers });
  }
  
  console.log("Simulación finalizada.");
}

startSimulation();
